File: 20171215/p06_distrib/random.c
#include <math.h>

// Random number generator; Uniform dist. in [0,1)
double random( int *seed ) {
// Input
//   seed    Integer seed (DO NOT USE A SEED OF ZERO)
// Output
//	 rand    Random number uniformly distributed in [0,1)

  const double a = 16807.0;
  const double m = 2147483647.0;
  double temp = a * (*seed);
  *seed = (long)(fmod(temp,m));
  double random = *seed/m;
  return( random );
}

File: 20171215/p06_distrib/random.f90
!--------------------------------------------------------------------------
!     Random number generator by the linear congruential method
!--------------------------------------------------------------------------
real(kind=8) function random( ix )
  real(kind=8) a /16807.d0/, c/0.0d0/,x
  real(kind=8)  M /2147483647d0/
  x      = ix
  x      = dmod( a*x+c,M )
  ix     = x
  random   = x/M

  return
end function random
